react-calculator/src/elements/Digit.js

```diff
@@ -1,20 +1,16 @@
-import React, { Component } from 'react';
+import React from "react";
 
-export default class Digit extends Component {
-  constructor(props) {
-    super(props);
-    this.state = { ...props };
-  }
-  render() {
-    return (
-      <button
-        className="digit"
-        onClick={() => {
-          this.state.onClickDigit(this.state.digit);
-        }}
-      >
-        {this.state.digit}
-      </button>
-    );
-  }
+function Digit({ digit, onClickDigit }) {
+  return (
+    <button
+      className="digit"
+      onClick={() => {
+        onClickDigit(digit);
+      }}
```

Comment:
이렇게 함수를 할당하게되면 컴포넌트가 마운트 될때 새로운 익명함수를 생성해서 할당해주기 때문에 상위 컴포넌트에서 그대로 넣어주면 더 좋을거같아요.

```suggestion
      onClick={onClickDigit(digit)}
```

```javascript
const onClickDigit = (digit) => () => {
  // ...
}
```